### PR TITLE
Issue #5041: Notice: Undefined index: form_build_id in ajax_get_form(…

### LINKED ISSUE #5041: Fix PHP Notice: Undefined index: form_build_id in ajax_get_form()
**Steps To Reproduce**

To reproduce the behavior just go to http://backdrop.site/system/ajax and the log will show the following notice:

> Notice: Undefined index: form_build_id in ajax_get_form() (line 323 of /var/www/docroot/core/includes/ajax.inc).

and the warning:

> Invalid form POST data.

**Additional information**

Normally, the `system/ajax` page should not be accessed directly. Regardless, an additional check could help to avoid the above warning and notice.

### PR DESCRIPTION
**Pull request by alanmels**
…) (line 323 of core/includes/ajax.inc).

Resolves https://github.com/backdrop/backdrop-issues/issues/5041

### CODE REVIEW

File: core/includes/ajax.inc [hunk 1]
```diff
@@ -318,6 +318,10 @@ function ajax_render($commands = array()) {
  *   @endcode
  */
 function ajax_get_form() {
+    // Exit early if $_POST is empty.
```
Comment (klonos) [inline core/includes/ajax.inc]:
```suggestion
  // Exit early if $_POST is empty.
```

File: core/includes/ajax.inc [hunk 2]
```diff
@@ -318,6 +318,10 @@ function ajax_render($commands = array()) {
  *   @endcode
  */
 function ajax_get_form() {
+  // Exit early if $_POST is empty.
+  if (empty($_POST)) {
```
Comment (indigoxela) [inline core/includes/ajax.inc]: One thought: form_build_id is essential in this function.

What if something gets posted (garbage stuff), but the form_build_id is still missing. Shouldn't we also check for that?

```
  if (empty($_POST) || !isset($_POST['form_build_id'])) {
    backdrop_exit();
  }
```
@alanmels what do you think?

Comment (alanmels) [reply core/includes/ajax.inc]: I've run checks many times and it's either `$_POST` is empty or it always contains `form_build_id`, so I thought checking the `$_POST` was sufficient. Can you confirm my finding?

Comment (indigoxela) [reply core/includes/ajax.inc]: I'm not sure, it was merely an idea based on the fact that nothing can be done in that function without the "form_build_id".

I'll leave it to you, to check for it or not. Checking for an empty $_POST will for sure fix most problems anyway.

Comment (indigoxela) [reply core/includes/ajax.inc]: Consider this discussion as resolved. :smile:

Comment (alanmels) [reply core/includes/ajax.inc]: Made the change, thanks for reviewing!

**Comment by quicksketch** [comment]
Thanks @alanmels! Merged into 1.x and 1.18.x.

**Pull request merged**